CONTRIBUTING.md

```diff
@@ -0,0 +1,69 @@
+# Contributing guide
+
+## API changes
+
+The following high level procedure is used to make Envoy changes that require API changes.
+
+1. Create a PR in this repo for the API/configuration changes. (If it helps to discuss the
+   configuration changes in the context of a code change, it is acceptable to point a code
+   change at a temporary fork of this repo so it passes tests).
+2. Bazel can be used to build/test locally.
+   1. Directly on Linux:
+
+      ```
+      bazel build //api/...
+      bazel test //test/... //tools/...
+      ```
+
+   2. Using docker:
+
+      ```
+      ./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.test'
+      ./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.docs'
+      ```
+3. All configuration changes should have temporary associated documentation. Fields should be
+   hidden from the documentation via the `[#not-implemented-hide:]` comment tag. E.g.,
+
+   ```
+   // [#not-implemented-hide:] Some new cool field that I'm going to implement and then
+   // come back and doc for real!
+   string foo_field = 3;
+   ```
```

Comment:
What about linking to PGV https://github.com/lyft/protoc-gen-validate/blob/master/validate/validate.proto and asking constraints to be specified?

Reply:
+1

Reply:
Ah ignore. I see it below